source/sema.h

```diff
@@ -19,23 +19,147 @@
 #define CPP2_SEMA_H
 
 #include "reflect.h"
+#include "dll.h"
 
 
 namespace cpp2 {
 
+static reflect_libraries* active_instance = nullptr;
+
+class reflect_libraries
+{
+    std::vector<error_entry>& errors;
+    std::string               in_progress = {};
+    std::vector<dll>          loaded      = {};
+
+    struct function_entry
+    {
+        std::variant<
+            std::monostate,
+            meta::mf_sign_type_in*,
+            meta::mf_sign_type_inout*
+        > type = {};
+        std::variant<
+            std::monostate,
+            meta::mf_sign_func_in*,
+            meta::mf_sign_func_inout*
+        > func = {};
+    };
+
+    std::map<std::string, function_entry> functions = {};
+
+public:
+    reflect_libraries(std::vector<error_entry>& errors_)
+        : errors{ errors_ }
+    { }
+
+    void load(std::string_view lib) {
+        active_instance = this;
+        in_progress = lib;
+        finally _{ [&](){ active_instance = nullptr; in_progress = ""; } };
+        auto& dl = loaded.emplace_back(in_progress);
+        if(!dl.is_open()) {
+            errors.emplace_back(
+                source_position(-1, -1),
+                "unable to load library '" + in_progress + "': " + dll::get_last_error()
+            );
+        }
+    }
+
+    bool try_apply(
+        [[maybe_unused]] std::string_view caller,
+        std::string const& name,
+        auto& decl
+    ) const {
+        // TODO(DyXel): Do full/better name look-up.
+        auto search = functions.find(name);
+        if(search == functions.end()) return false;
+        auto& entry = search->second;
```

Comment:
Leave an empty line here. The early out is hardly recognizeable. Maybe also add a comment.

Reply:
Thanks for pointing it out, it could be great to have #1253 so I can stop worrying about stylistic choices. In any case, this code is throwaway since I need to implement full registration and look-up.